specs/interop/shared-lockbox.md

```diff
@@ -0,0 +1,340 @@
+  - [Contracts upgrades](#contracts-upgrades)
+    - [`SuperchainConfig`](#superchainconfig)
+    - [`OptimismPortal`](#optimismportal)
+- [Reference implementation](#reference-implementation)
+  - [`SharedLockbox`](#sharedlockbox-1)
+  - [`LiquidityMigrator`](#liquiditymigrator-1)
+  - [`SuperchainConfig`](#superchainconfig-1)
+  - [`OptimismPortal`](#optimismportal-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+With interoperable ETH, withdrawals may fail if the referenced `OptimismPortal` lacks sufficient ETH
+— making it mandatory to withdraw from the L2 where the ETH was originally deposited.
+The `SharedLockbox` improves the Superchain's interoperable ETH withdrawal user experience and avoid this issue.
+To do so, it unifies ETH L1 liquidity in a single contract, enabling seamless withdrawals of ETH from any OP chain
+in the Superchain, regardless of where the ETH was initially deposited.
+
+## Design
+
+### Contracts inroductions
+
+#### `SharedLockbox`
+
+The `SharedLockbox` contract is designed to manage the unified ETH liquidity for the Superchain.
+It provides two main functions: `lockETH` for depositing ETH into the lockbox,
+and `unlockETH` for withdrawing ETH from the lockbox.
+These functions are called by the `OptimismPortal` contracts to manage the shared ETH liquidity
+when making deposits or finalizing withdrawals.
+
+**Interface and properties:**
+
+**`lockETH`**
+
+Deposits and locks ETH into the lockbox's liquidity pool.
+
+- The function MUST accept ETH.
+- Only authorized `OptimismPortal` addresses MUST be allowed to interact.
+- The function MUST emit the `ETHLocked` event with the correct `portal` and `amount`.
+
+```solidity
+function lockETH() external payable;
+```
+
+**`unlockETH`**
+
+Withdraws a specified amount of ETH from the lockbox's liquidity pool.
+
+- Only authorized `OptimismPortal` addresses MUST be allowed to interact.
+- The function MUST emit the `ETHUnlocked` event with the correct `portal` and `amount`.
+
+```solidity
+function unlockETH(uint256 _value) external;
+```
+
+**`authorizePortal`**
+
+Grants authorization to a specific `OptimismPortal` contract.
+
+- Only `SuperchainConfig` address MUST be allowed to interact.
+- The function MUST add the specified address to the mapping of authorized portals.
+- The function MUST emit the `AuthorizedPortal` event when a portal is successfully added.
+
+```solidity
+function authorizePortal(address _portal) external;
+```
+
+**`ETHLocked`**
+
+MUST trigger when `lockETH` is called
+
+```solidity
+event ETHLocked(address indexed portal, uint256 amount);
+```
+
+**`ETHUnlocked`**
+
+MUST trigger when `unlockETH` is called
+
+```solidity
+event ETHUnlocked(address indexed portal, uint256 amount);
+```
+
+**`AuthorizedPortal`**
+
+MUST trigger when `authorizePortal` is called
+
+```solidity
+event AuthorizedPortal(address indexed portal);
+```
+
+#### `LiquidityMigrator`
```

Comment:
I find beneficial to explain why using this instead of just creating a one time function or executing this logic on `initialize()` inside the Portal?